src/shared.ts

```diff
@@ -66,3 +58,11 @@ export function registerProxy(proxy: object, value: any) {
 }
 
 export const unwrap = (replicaOrAny: any): any => proxyToValueMap.get(replicaOrAny) || replicaOrAny;
+
+export function getOwnKeysAndMagicProperty(target: object, magicPropertyKey: PropertyKey | undefined): PropertyKey[] {
+    const keys = isUndefined(magicPropertyKey) || hasOwnProperty.call(target, magicPropertyKey) ? [] : [magicPropertyKey];
```

Comment:
`isUndefined` returns a boolean. I'm not sure if `const keys = isUndefined(magicPropertyKey)` can be problematic.

```suggestion
    if (isUndefined(magicPropertyKey)) {
        return undefined;
    }
    const keys = hasOwnProperty.call(target, magicPropertyKey) ? [] : [magicPropertyKey];
```

Particularly, I don't love ternaries so I'd also avoid them:

```suggestion
    if (isUndefined(magicPropertyKey)) {
        return undefined;
    }
    const keys = [];
    if (!hasOwnProperty.call(target, magicPropertyKey)) {
      ArrayPush.apply(keys, magicPropertyKey);
    }
```

Problem here is the `!`, but it could be replaced with a trailing `=== false`

Reply:
there are probably some gaps here... but based on the signature of this, you must return an array, not undefined... and that array should alwyas contain all the own properties of the target, and one extra property, the magic property, if the target doesnt' have it.

Reply:
I really got confused with the ternary. Now I see it gets the `[]` if magicPropertyKey is undefined.